tensorflow/lite/core/subgraph.cc

```diff
@@ -424,44 +424,69 @@ TfLiteStatus Subgraph::CheckTensorIndices(const char* label, const int* indices,
   return kTfLiteOk;
 }
 
+int64_t Subgraph::MultiplyWithoutOverflow(const int64_t x, const int64_t y) {
+  // Multiply in uint64 rather than int64 since signed overflow is undefined.
+  // Negative values will wrap around to large unsigned values in the casts
+  // (see section 4.7 [conv.integral] of the C++14 standard).
+  const uint64_t ux = x;
+  const uint64_t uy = y;
+  const uint64_t uxy = ux * uy;
+
+  // Check if we overflow uint64, using a cheap check if both inputs are small
+  if (((ux | uy) >> 32) != 0) {
+    // Ensure nonnegativity.  Note that negative numbers will appear "large"
+    // to the unsigned comparisons above.
+    TF_LITE_ENSURE(context_, x >= 0 && y >= 0);
```

Comment:
I'm not sure we should return the error as an int64_t here? As that will be ignored?

Either this method should return an error, and the int64_t result is a pointer arg, or we should return -1 here.